plugin/test-keeper/plugin/event_handler_test.go

```diff
@@ -0,0 +1,180 @@
+package plugin_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/arquillian/ike-prow-plugins/plugin/test-keeper/plugin"
+	"gopkg.in/h2non/gock.v1"
+	"github.com/sirupsen/logrus"
+	"github.com/arquillian/ike-prow-plugins/plugin/github"
+	"os"
+	"io/ioutil"
+	"fmt"
+	"github.com/google/go-github/github"
+	"net/http"
+	"encoding/json"
+	"io"
+)
+
+const eventGUID = "event-guid"
+
+var _ = Describe("Test Keeper Plugin features", func() {
+
+	var logger *logrus.Entry
+
+	BeforeSuite(func() {
+		nullLogger := logrus.New()
+		nullLogger.Out = ioutil.Discard // TODO rethink if we want to discard logging entirely
+		logger = logrus.NewEntry(nullLogger)
+	})
+
+	Context("Pull Request handling", func() {
+
+		var handler *plugin.GitHubTestEventsHandler
+
+		BeforeEach(func() {
+			defer gock.Off()
+
+			client := github.NewClient(nil) // TODO with hoverfly/go-vcr we might want to use tokens instead to capture real traffic
+			handler = &plugin.GitHubTestEventsHandler{
+				Client: client,
+				Log:    logger,
+			}
+		})
+
+		It("should approve opened pull request when tests included", func() {
+			// given
+			gock.New("https://api.github.com").
+				Get("/repos/bartoszmajsak/wfswarm-booster-pipeline-test/pulls/2/files").
+				Reply(200).
+				Body(fromJson("test_fixtures/github_calls/prs/with_tests/changes.json"))
+
+			toHaveSuccessState := func(statusPayload map[string]interface{}) (bool) {
```

Comment:
these functions `toHaveSuccessState` and `toHaveFailureState` can be as a global variables

Reply:
Would it make the test more readable? You will have to jump between those. It's violating DRY, but keeps test's "story" in one place. I think about it as an assertion and like with those I tend not to extract it outside of the test.

Reply:
Are you fine with leaving it here or have strong preference on moving it to global variable?

Reply:
I don't insist on it. Let's keep it there and if the file grows, then we can think about making it as a global variable to keep the file shorter.

Reply:
![agree](https://user-images.githubusercontent.com/719616/36989088-f7f59af4-20a0-11e8-9f20-40ca8d140b30.gif)